File: fornituras-api/src/main/java/com/numobiz/solutions/fornituras/modules/officers/service/PiiMasker.java
package com.numobiz.solutions.fornituras.modules.officers.service;

/**
 * Enmascara identificadores de PII (CURP/RFC) para roles no autorizados: muestra los primeros
 * caracteres y oculta el resto, de modo que un operador pueda cotejar parcialmente sin ver el dato
 * completo. La decisión de enmascarar la toma el servidor (nunca el cliente).
 */
public final class PiiMasker {

	private static final int VISIBLE_PREFIX = 4;

	private PiiMasker() {
	}

	public static String mask(String value) {
		if (value == null || value.isBlank()) {
			return value;
		}
		String trimmed = value.trim();
		if (trimmed.length() <= VISIBLE_PREFIX) {
			return "•".repeat(trimmed.length());
		}
		String prefix = trimmed.substring(0, VISIBLE_PREFIX);
		return prefix + "•".repeat(trimmed.length() - VISIBLE_PREFIX);
	}
}

File: fornituras-api-dotnet/src/Fornituras.Api/Services/PiiMasker.cs
namespace Fornituras.Api.Services;

/// <summary>
/// Enmascara identificadores de PII (CURP/RFC): primeros 4 caracteres visibles.
/// </summary>
public static class PiiMasker
{
    private const int VisiblePrefix = 4;

    public static string? Mask(string? value)
    {
        if (string.IsNullOrWhiteSpace(value))
        {
            return value;
        }

        var trimmed = value.Trim();
        if (trimmed.Length <= VisiblePrefix)
        {
            return new string('•', trimmed.Length);
        }

        return trimmed[..VisiblePrefix] + new string('•', trimmed.Length - VisiblePrefix);
    }
}
